File: Mona/src/lib/java/Math.cpp
// This file was created by `gcjh -stubs'. -*- c++ -*-
//
// This file is intended to give you a head start on implementing native
// methods using CNI.
// Be aware: running `gcjh -stubs ' once more for this class may
// overwrite any edits you have made to this file.

#include <gcj/javaprims.h>
#include <java/lang/Math.h>
#include <gcj/cni.h>
#ifdef MONA
#include <monapi.h>
#else
#include <math.h>
#endif

jdouble
java::lang::Math::sqrt (jdouble a)
{
	return ::sqrt(a);
}


jdouble
java::lang::Math::sin (jdouble a)
{
	return ::sin(a);
}


jdouble
java::lang::Math::cos (jdouble a)
{
	return ::cos(a);
}


jdouble
java::lang::Math::tan (jdouble a)
{
	return ::tan(a);
}


jdouble
java::lang::Math::atan (jdouble a)
{
	return ::atan(a);
}


jdouble
java::lang::Math::atan2 (jdouble y, jdouble x)
{
	return ::atan2(y, x);
}


jdouble
java::lang::Math::pow (jdouble a, jdouble b)
{
	return ::pow(a, b);
}

File: contrib/Devel/libjava/java/lang/Math.java
package java.lang;

public class Math {
	public static final double PI = 3.14159265358979;

	public static int abs(int a) {
		return a >= 0 ? a : -a;
	}
	public static native double sqrt(double a);
	public static native double sin(double a);
	public static native double cos(double a);
	public static native double tan(double a);
	public static native double acos(double a);
	public static native double atan(double a);
	public static native double atan2(double y, double x);
	public static native double pow(double a, double b);

	private static double seed;
	public static double random() {
		if (seed == 0) seed = PI * 0.1;
		seed *= 17;
		seed -= (double)(int)seed;
		return seed;
	}
}
